manager/catalog.go

```diff
@@ -178,7 +191,24 @@ func (cat *Catalog) walkCatalog(path string, f os.FileInfo, err error) error {
 func (cat *Catalog) pullCatalog() error {
 	log.Debugf("Pulling the catalog %s from the repo to sync any new changes to %s", cat.CatalogID, cat.catalogRoot)
 
-	e := exec.Command("git", "-C", cat.catalogRoot, "pull", "-r", "origin", "master")
+	// git --git-dir=./DATA/value/.git/ --work-tree=./DATA/value/ checkout new_branch
+	gitCheckoutCmd := exec.Command("git", "--git-dir="+cat.catalogRoot+"/.git", "--work-tree="+cat.catalogRoot, "checkout", cat.URLBranch)
+
+	out, gitCheckoutErr := gitCheckoutCmd.Output()
+	if gitCheckoutErr != nil {
+		errorStr := "Git checkout failure from git err: " + gitCheckoutErr.Error()
+		log.Error(errorStr)
+	}
+	log.Debugf("Branch to be worked on : %s\n", out)
+
+	var e *exec.Cmd
+	if cat.URLBranch != "master" {
```

Comment:
oh, I had some other command as well in the else block previously, I will remove the if-else